src/westeros/WesterosViewbackendInput.cpp

```diff
@@ -144,7 +144,7 @@ void WesterosViewbackendInput::handleKeyEvent(void* userData, uint32_t key, uint
 
         uint32_t keysym = wpe_input_xkb_context_get_key_code(wpe_input_xkb_context_get_default(), e->key, e->state == WL_KEYBOARD_KEY_STATE_PRESSED);
         if (!keysym)
-            return;
+            return 0;
```

Comment:
What is the intention here? My understanding is that we always want to remove the idle source after the callback is triggered, and that's why you are returning zero (which is a “falsey” value). Please use `return G_SOURCE_REMOVE` instead for clarity.